tests/rptest/tests/self_test_test.py

```diff
@@ -29,7 +29,10 @@ class SelfTestTest(RedpandaTest):
     ]
 
     def __init__(self, ctx):
-        super(SelfTestTest, self).__init__(test_context=ctx)
+
+        super(SelfTestTest,
+              self).__init__(test_context=ctx,
+                             si_settings=SISettings(test_context=ctx))
```

Comment:
Isn't this now only testing one case: where tiered storage is fully enabled? Could you add some coverage for some more interesting cases, e.g. reads and/or writes disabled, S3 down, etc. It'd also be nice to leave a run of this test that doesn't use tiered storage at all. To that end, consider using `parametrize`.

Though they don't all have to be a part of this same test (e.g. maybe it makes sense to write a new fixture test that just exercises the `cloudcheck` backed by an `s3_imposter`?)

Reply:
Parameterized ducktape test for all read/write permissions.